CRM/Utils/Money.php

```diff
@@ -85,8 +85,8 @@ public static function format($amount, $currency = NULL, $format = NULL, $onlyNu
     if ($currency === ' ') {
       CRM_Core_Error::deprecatedWarning('Passing empty currency to CRM_Utils_Money::format is deprecated if you need it for display without currency call CRM_Utils_Money::formatLocaleNumericRounded');
     }
-
-    $amount = self::formatNumericByFormat($amount);
+    $precision = self::getDecimalPlacesForAmount((string) $amount);
+    $amount = self::formatUSLocaleNumericRounded($amount, $precision > 2 ? $precision : 2);
```

Comment:
Why are we hardcoding a minimum of 2 decimal places here? There are quite a few currencies that have 0 decimal places - see eg. https://developer.cybersource.com/library/documentation/sbc/quickref/currencies.pdf

Reply:
Setting 0 instead of 2 would be ok here?

Reply:
Yeah - oddly enough there are still tests failing with this change - suggesting that we were rounding to 2 decimal places more often than it seemed - despite @demeritcowboy's test showing us not doing so ....

Reply:
It's actually the line

```
     $amount = money_format( '%!i', $amount);
```

that is doing the rounding

Reply:
Oh, of course, there is no money_format on windows.

Reply:
@demeritcowboy hmm - so rounding to 2 places would be consistent with the intent of the code I believe

Reply:
Yes it seems. On one unix machine I'm looking at it appears to be 2 places using, oddly, HALF_DOWN.